src/audio/module_adapter/module/volume/volume_generic_with_peakvol.c

```diff
@@ -144,7 +147,8 @@ static void vol_s32_to_s32(struct processing_module *mod, struct input_stream_bu
 							   Q_SHIFT_BITS_64(31, VOL_QXY_Y, 31));
 				tmp = MAX(abs(x0[i]), tmp);
 			}
-			cd->peak_regs.peak_meter[j] = MAX(tmp, cd->peak_regs.peak_meter[j]);
+			tmp = MAX(tmp, cd->peak_regs.peak_meter[j]);
+			cd->peak_regs.peak_meter[j] = tmp << attenuation;
```

Comment:
does the calculation right? should there be cd->peak_regs.peak_meter[j] = MAX(tmp<< attenuation, cd->peak_regs.peak_meter[j]);?